packages/block-editor/src/components/inserter/utils.js

```diff
@@ -1,45 +1,71 @@
+// TODO jsdoc + tests
+const getVariationMapper = ( item ) => ( variation ) => ( {
+	...item,
+	id: `${ item.id }-${ variation.name }`,
+	icon: variation.icon || item.icon,
+	title: variation.title || item.title,
+	description: variation.description || item.description,
+	// If `example` is explicitly undefined for the variation, the preview will not be shown.
+	example: variation.hasOwnProperty( 'example' )
+		? variation.example
+		: item.example,
+	initialAttributes: {
+		...item.initialAttributes,
+		...variation.attributes,
+	},
+	innerBlocks: variation.innerBlocks,
+} );
+
 /**
  * Normalizes an inserter block types list and includes variations as separate items.
  *
  * @param {Array} items Denormalized inserter items
+ * @param {number} [limit] Limit returned results by a given threshold. If limit===-1 return all
  * @return {Array} Normalized inserter items.
  */
-export function includeVariationsInInserterItems( items ) {
-	return items.reduce( ( result, item ) => {
-		const { variations = [] } = item;
-		const hasDefaultVariation = variations.some(
-			( { isDefault } ) => isDefault
-		);
+export function includeVariationsInInserterItems( items, limit = -1 ) {
```

Comment:
1. There's some inconsistency between this default of `-1` and the subsequent checks against `0`. Since this isn't an `index`-type function, I suggest `0` instead of `-1`. Another option totally valid in JS would be `limit = Infinity`, which is kinda cool.

2. The inclusion check below reads `limit > 0` but its counterpart reads `limit < 1`. To avoid surprises, I recommend `limit <= 0` (or just `limit === 0`, `! limit`, whichever you prefer, since we should move away from the `-1` default).

Reply:
> this default of -1
>  would be limit = Infinity

I had thought about `Infinity` too but I used `-1` with `$posts_per_page` in my mind. Let's go with `Infinity` then.

Regarding your second comment, the checks will change.